keanu-project/src/main/java/io/improbable/keanu/util/io/DotSaver.java

```diff
@@ -48,56 +46,35 @@ public DotSaver(BayesianNetwork network) {
     }
```

Comment:
`bayesianNetwork` is only used once, to `getAllVertices`.

If you replace the `BayesianNetwork` field with a `Set<Vertex>` field, then the user still has the option to construct the `DotSaver` with `bayesianNetwork.getSubgraph(vertex, degree)` if they wish.

Reply:
You can also keep the constructor `DotSaver(BayesianNetwork)` as a convenience if you like.

    public DotSaver(BayesianNetwork network) {
        this(network.getAllVertices);
    }